src/lib/yearn-affiliate-wrapper.sol

```diff
@@ -0,0 +1,357 @@
+        //       number of shares credited, which helps avoid an occasional multiplication
+        //       overflow if trying to adjust the number of shares by the share price.
+        uint256 beforeBal = token.balanceOf(address(this));
+        if (receiver != address(this)) {
+            _bestVault.deposit(amount, receiver);
+        } else if (amount != DEPOSIT_EVERYTHING) {
+            _bestVault.deposit(amount);
+        } else {
+            _bestVault.deposit();
+        }
+
+        uint256 afterBal = token.balanceOf(address(this));
+        deposited = beforeBal.sub(afterBal);
+        // `receiver` now has shares of `_bestVault` as balance, converted to `token` here
+        // Issue a refund if not everything was deposited
+        if (depositor != address(this) && afterBal > 0) token.transfer(depositor, afterBal);
+    }
+
+    function _withdraw(
+        address sender,
+        address receiver,
+        uint256 amount, // if `MAX_UINT256`, just withdraw everything
+        bool withdrawFromBest // If true, also withdraw from `_bestVault`
+    ) internal returns (uint256 withdrawn) {
+        VaultAPI _bestVault = bestVault();
+
+        VaultAPI[] memory vaults = allVaults();
+        _updateVaultCache(vaults);
+
+        for (uint256 id = 0; id < vaults.length; id++) {
+            if (!withdrawFromBest && vaults[id] == _bestVault) {
+                continue; // Don't withdraw from the best
+            }
+
+            // Start with the total shares that `sender` has
+            uint256 availableShares = vaults[id].balanceOf(sender);
+
+            // Restrict by the allowance that `sender` has to this contract
+            // NOTE: No need for allowance check if `sender` is this contract
+            if (sender != address(this)) {
+                availableShares = Math.min(availableShares, vaults[id].allowance(sender, address(this)));
+            }
+
+            // Limit by maximum withdrawal size from each vault
+            availableShares = Math.min(availableShares, vaults[id].maxAvailableShares());
+
+            if (availableShares > 0) {
+                // Intermediate step to move shares to this contract before withdrawing
+                // NOTE: No need for share transfer if this contract is `sender`
+                if (sender != address(this)) vaults[id].transferFrom(sender, address(this), availableShares);
+
+                if (amount != WITHDRAW_EVERYTHING) {
+                    // Compute amount to withdraw fully to satisfy the request
+                    uint256 estimatedShares = amount
+                        .sub(withdrawn) // NOTE: Changes every iteration
+                        .mul(10**uint256(vaults[id].decimals()))
+                        .div(vaults[id].pricePerShare()); // NOTE: Every Vault is different
+
+                    // Limit amount to withdraw to the maximum made available to this contract
+                    uint256 shares = Math.min(estimatedShares, availableShares);
```

Comment:
There also have been some updates on this piece of code on latest from master

https://github.com/yearn/yearn-vaults/blob/c818595d1699077d20285c5fdd28ed2a9f564b6c/contracts/BaseWrapper.sol#L239